src/spec/PlayerHand.test.ts

```diff
@@ -0,0 +1,16 @@
+import { PlayerHand } from '../types/PlayerHand';
+import * as O from 'fp-ts/lib/Option'
+
+test('piece selection', () => {
+    const hand = new PlayerHand();
+    expect(hand.getSelectedPiece()).toBe(O.none)
+
+    const selected = hand.setSelectedPiece('piece1');
+    expect(selected.getSelectedPiece()).toStrictEqual(O.some({ name: 'piece1', status: 'selected' }))
+
+    const selected2 = selected.setSelectedPiece('piece4');
+    expect(selected2.getSelectedPiece()).toStrictEqual(O.some({ name: 'piece4', status: 'selected' }))
+
+    const cleared = selected2.clearSelectedPiece();
+    expect(cleared.getSelectedPiece()).toBe(O.none)
+});
```

Comment:
I could have added some more tests.
I'm expecting these classes to change and haven't yet decided on interfaces that these should implement.
These tests are a bit brittle and the classes currently are a bit of a leaky abstraction.
Once I get into the logic and representation for the pieces, board, and game rules I should finalize some interfaces for these and write better tests to the implementation of the interfaces as opposed to the internal representation